### PR TITLE
feat: change to generic API endpoint.

### PR DESCRIPTION
**Pull request by gagik**
https://app.asana.com/0/1201443343334931/1201637760640810

### CODE REVIEW

File: pages/app/index.tsx [hunk 1]
```diff
@@ -7,7 +7,7 @@ export default withPageAuthRequired(() => {
   const [errorMessage, setErrorMessage] = useState("")
   const [success, setSuccess] = useState(false)
   const sendMail = async (mailData) => {
-    const response = await fetch('/api/mail/send', {
+    const response = await fetch('/api/hoagie/mail/send', {
```
Comment (gagik) [inline pages/app/index.tsx:10]: Should I use "hoagie" for the URL here?

File: pages/api/hoagie/[...query].tsx [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        headers: {
+            'Content-Type': 'application/json',
+            Authorization: `Bearer ${accessToken}`,
+        },
+        body: req.body,
+    }
+
+    await fetch(`${process.env.HOAGIE_API_URL}/${query}`, fetchReq)
+        .then(async (response) => {
+            if (response.status === 204) {
```
Comment (gagik) [inline pages/api/hoagie/[...query].tsx:29]: I am not sure whether I should hardcode 204 here.

**Review by gagik — COMMENTED** [review]
Just some things I am unsure about...

File: pages/app/index.tsx (hunk 1 above)
Comment (djapp18) [reply pages/app/index.tsx:10]: Yeah looks fine

File: pages/api/hoagie/[...query].tsx (hunk 2 above)
Comment (djapp18) [reply pages/api/hoagie/[...query].tsx:29]: Just out of curiosity what functionality would require 204

**Review by djapp18 — COMMENTED** [review]
overall looks fine see comments for general clarification

**Pull request merged**